### PR TITLE
Remove Unused Process in __builtin:item

### PR DESCRIPTION
**Pull request by Rui914**

### CODE REVIEW

File: builtin/game/item_entity.lua [hunk 1]
```diff
@@ -46,16 +43,7 @@ core.register_entity(":__builtin:item", {
 		local s = 0.2 + 0.1 * (count / max_count)
 		local c = s
 		local itemtable = stack:to_table()
-		local itemname = nil
-		if itemtable then
-			itemname = stack:to_table().name
-		end
-		local item_texture = nil
```
Comment (est31) [inline builtin/game/item_entity.lua]: why do you remove this?

File: builtin/game/item_entity.lua [hunk 2]
```diff
@@ -149,7 +135,7 @@ core.register_entity(":__builtin:item", {
 	on_step = function(self, dtime)
 		self.age = self.age + dtime
 		if time_to_live > 0 and self.age > time_to_live then
-			self.itemstring = ''
+			self.itemstring = ""
```
Comment (est31) [inline builtin/game/item_entity.lua]: why not remove this too?

**Comment by est31** [comment]
While I agree to some of the cleanups, some code might have been done in order to not make it abuseable that you can punch an item multiple times, and get the stack multiple times as well, I'm not sure. Can you find out whether this is the case?

**Comment by Ekdohibs** [comment]
The "self.itemstring =''" is needed when removing the object due to the inner workings of self.object:remove: indeed, the object will only be deleted after returning from Lua, and other mods have no way to see that the object has been deleted, see https://github.com/minetest/minetest/commit/936c6f577ad70c9a467eaed58233e737671375df

**Comment by est31** [comment]
Perhaps a comment can be added to ensure future readers of the code don't remove it.

File: builtin/game/item_entity.lua (hunk 2 above)
Comment (Rui914) [inline builtin/game/item_entity.lua]: Just a forgot to erase...

But, now I understand I shouldn't remove this.

File: builtin/game/item_entity.lua (hunk 1 above)
Comment (Rui914) [inline builtin/game/item_entity.lua]: Because this is unused.

...don't I perhaps mistaken?

File: builtin/game/item_entity.lua (hunk 2 above)
Comment (Rui914) [inline builtin/game/item_entity.lua]: Because just forgot to erase...

But, I understand I shouldn't remove this.

**Comment by Ekdohibs** [comment]
Yes, that would be a good thing to do, I will add it if Rui doesn't do it with this pull.

**Pull request closed without merging**